src/components/network-modification-tree-pane.js

```diff
@@ -138,8 +140,21 @@ export const NetworkModificationTreePane = ({
 
     const handleCreateNode = useCallback(
         (element, type, insertMode) => {
+            getUniqueNodeName(studyUuid)
+                .then((response) => setNewName(response))
+                .catch((errorMessage) => {
+                    displayErrorMessageWithSnackbar({
+                        errorMessage: errorMessage,
+                        enqueueSnackbar: enqueueSnackbar,
+                        headerMessage: {
+                            headerMessageId: 'NodeCreateError',
+                            intlRef: intlRef,
+                        },
+                    });
+                });
+
             createTreeNode(studyUuid, element.id, insertMode, {
```

Comment:
You should call `createTreeNode` in the `then()` proc of the `getUniqueNodeName`.
Because here you don't wait for the response to create the new node with the correct name. Then you could even remove unecessary `newName` state